docs/setup/custom-config.mdx

```diff
@@ -95,33 +149,71 @@ First, you should prepare a `config.json` file based on the following template;
+* `ssl` (for MySQL API): Indicates whether SSL support is enabled for the MySQL API.
+
+```bash
     "cache": {
-        "type": "local"
+        "type": "local",
+        "connection": "redis://localhost:6379" # optional
     },
+```
+
+The `cache` parameter controls how MindsDB stores the results of recent predictions to avoid recalculating them if the same query is run again. Note that recent predictions are cached for ML models, like Lightwood, but not in the case of large language models (LLMs), like OpenAI.
+
+The `type` parameter specifies the type of caching mechanism to use for storing prediction results.
+
+* `none`: Disables caching. No prediction results are stored.
+* `local` (default): Stores prediction results in the `cache` folder (as defined in the `paths` configuration). This is useful for repeated queries where the result doesn't change.
+* `redis`: Stores prediction results in a Redis instance. This option requires the `connection` parameter, which specifies the Redis connection string.
+
+```bash
     "ml_task_queue": {
-        "type": "local"
+        "type": "local",
+        "host": "localhost", # required only when type is set to redis
+        "port": 6379, # required only when type is set to redis
+        "db": 0, # required only when type is set to redis
+        "username": "username", # required only when type is set to redis
+        "password": "password" # required only when type is set to redis
     },
-    "file_upload_domains": [],
-    "web_crawling_allowed_sites": [],
-}
 ```
 
-<Tip>
-Configure HTTP authentication for your MindsDB instance by setting the environment variables `MINDSDB_USERNAME` and `MINDSDB_PASSWORD` before starting MindsDB. Alternatively, you can also enable authentication by setting `http_auth_enabled` to `True` and defining the `username` and `password` parameters in the `auth` section of the above configuration file, however, this is not recommended for production environments.
-</Tip>
+The `ml_task_queue` parameter manages the queueing system for machine learning tasks in MindsDB. ML tasks include operations such as creating, training, predicting, fine-tuning, and retraining models. These tasks can be resource-intensive, and running multiple ML tasks simultaneously may lead to Out of Memory (OOM) errors or performance degradation. To address this, MindsDB uses a task queue to control task execution and optimize resource utilization.
+
+The `type` parameter defines the type of task queue to use.
 
-Now that your `config.json` file is ready, run the command below to start
-MindsDB locally with your custom configuration.
+* `local`: Tasks are processed immediately as they appear, without a queue. This is suitable for environments where resource constraints are not a concern.
+* `redis`: Tasks are added to a Redis-based queue. Redis ensures that tasks are executed only when sufficient resources are available.
```

Comment:
`Redis ensures that tasks are executed only when sufficient resources are available.` Consumer process (which you run with --ml_task_consumer) do it, not the redis.

Reply:
Updated it to:
> * `redis`: Tasks are added to a Redis-based queue, and consumer process (which is run with `--ml_task_consumer`) ensures that tasks are executed only when sufficient resources are available.